dll/win32/shdocvw/mrulist.cpp

```diff
@@ -32,11 +33,13 @@ class CMruBase;
             class CMruPidlList;
 class CMruClassFactory;
 
-extern "C" void __cxa_pure_virtual(void)
+#ifdef __GNUC__ // GCC and Clang.
+extern "C" void __cxa_pure_virtual()
```

Comment:
What's happening with MSVC builds then?
Do we also know why those things were needed only in these three C++ modules, and not all the others?

Reply:
> What's happening with MSVC builds then?

Per a web search, it looks like `__cxa_pure_virtual()` is GCC (and Clang) specific.
Compilation with MSVC (and Clang-cl) succeed, which would seem to confirm that..

>  Do we also know why those things were needed only in these three C++ modules, and not all the others?

I have no idea: I am simply sync'ing (and narrowing) existing code.

You might ask authors:

mrulist.cpp: 2023-08-31
https://github.com/reactos/reactos/commit/58770200e75bd8f0c44485b0281fbd97fa799406#diff-139e6c11bf066b06e72abbd5e181a27e0bdfeadd426c8ece544a94096a8f8f61

purecall.cpp : r43026
https://github.com/reactos/reactos/commit/ad2cf72549b9e95e537ff9291cd8187ae4948d38#diff-2986c8a086e3bd8f84009939250e7a788e7c9b852b7137aa8d41309df1f4854d

cunknown.cpp : r58641
https://github.com/reactos/reactos/commit/a08dba228ca0bf3a63c3e27ae59325b51af780cd#diff-b53cfcab95e7a76d75b6887940184462cebe6f8ff08cf4f0109fea547b969dbc

reactos_special.cpp: 2020-11-02
https://github.com/reactos/reactos/commit/1f377076d783d597f33e720498ef29505973c680#diff-062c49329af0b4110974cfa6b07a88b4ae329bb1b1789fab77482e29b16f54bd